.github/jobs/BuildNativeJob.pkl

```diff
@@ -0,0 +1,85 @@
+extends "GradleJob.pkl"
+
+import "@gha/actions/Artifact.pkl"
+
+/// Whether to link to musl. Otherwise, links to glibc.
+musl: Boolean(implies(module.os == "linux")) = false
+
+/// The Gradle project under which to generate the executable
+project: String
+
+javaVersion = "21.0"
+
+extraGradleArgs {
+  when (os == "macOS" && arch == "amd64") {
+    "-Dpkl.targetArch=\(module.arch)"
+  }
+  when (musl) {
+    "-Dpkl.musl=true"
+  }
+}
+
+local muslVersion = "1.2.5"
+local zlibVersion = "1.2.13"
+
+steps {
+  when (musl) {
+    new {
+      name = "Install musl and zlib"
+      // language=bash
+      run =
+        #"""
+        mkdir -p ~/staticdeps/
+
+        # install zlib
+        if [[ ! -f ~/staticdeps/include/zlib.h ]]; then
+          curl -Lf https://github.com/madler/zlib/releases/download/v\#(zlibVersion)/zlib-\#(zlibVersion).tar.gz -o /tmp/zlib.tar.gz
```

Comment:
Can we add gpg signature verification for both zlib and musl? They both provide .asc files alongside their releases.